doc/ni_by_example.md

```diff
@@ -0,0 +1,564 @@
+
+To print fields 11-15 of a data source, you would use `$ ni ... r F_ 10..14`.
+
+The index of the final nonempty field in a line is stored in `FM`. To get the last field of every line in our example, you could use the spell: `$ ni /usr/share/dict/words rx40 r10 p'r substr(a, 0, 3), substr(a, 3, 3), substr(a, 6)' p'r FM, F_ FM'`
+
+It is often useful to take everything after a certain point in a line. This can be accomplished efficiently using the `FR n` operator. `FR n` is equivalent to `F_ n..FM`.
+
+
+
+##Sort, Unique, and Count
+Sorting large amounts of data requires buffering to disk, and the vagaries of how this works internally is a source of headaches and slow performance. If your data is larger than a gigabyte uncompressed, you may want to take advantage of massively distributing the workload through Hadoop operations.
+
+If you're not convinced that anything could go slow in `ni`, let's try counting all of the letters in the dictionary 
+
+`$ ni /usr/share/dict/words F// p'FR 0' gc \>letter_counts.txt`
+
+You'll probably see the `ni` monitor for the first time, showing the steps of the computation, and what steps are taking time.  The whole process takes about 90 seconds on my computer.
+
+```
+$ ni --explain /usr/share/dict/words F// p'FR 0' gc \>letter_counts.txt
+ni --explain /usr/share/dict/words F// p'FR 0' gc \>letter_counts.txt
+["cat","/usr/share/dict/words"]
+["split_regex",""]
+["perl_mapper","FR 0"]
+["row_sort","-t","\t"]
+["count"]
+["file_write","letter_counts.txt"]
+```
+
+What's interesting here is that your Unix dictionary is probably only about 2.5 MB in size; the dictionary itself can be streamed into memory in a fraction of a second. Remembering that all `ni` development code such that it is I/O bounded; in this case, there is an I/O bounded step where the data must be written to disk in order to be sorted. One can avoid this bound by adding `r10` after the filename, however.
+
+
+#### `g`: General sorting
+`g` is the most general sorting operator; there are two other sorting operators within `ni`, but they are highly specific.
+
+With a single column of data, as in the example, the simple command `g` will give you lexicographic sorting **in ASCII, in ascending order**. 
+
+**Feature test: Unicode**
+
+To do more complicated sorts, you can give `g` columns and modifiers. As with `f`, columns are indexed `A-Z`. `g` has two modifiers, `n` and `-`. `n` makes the sort numeric, and `-` makes the sort descending, rather tahn ascending.
+
+Examples: 
+
+`$ ni /usr/share/dict/words F// p'FR 0' gc =\>letter_counts.txt gAn =\>ascending_letter_counts.txt gB- \>counts_by_letter_reversed.txt`
+
+As above, if you have more than one column, you currently **must** specify the columns you want sorted; the reason for this is a system-to-system instability with regard to how the unix `sort` interacts).
+  
+####`c`: Count Sorted Rows
+`c` is `ni` syntax for `uniq -c`, which counts the number of unique rows.
```

Comment:
The output is also subtly different from `uniq -c`; `uniq` doesn't tab-delimit between the count and the value (which is why ni implements its own version)